core/templates/pages/contributor-dashboard-page/services/translate-text-backend-api.service.ts

```diff
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// eslint-disable-next-line oppia/no-multiline-disable
```

Comment:
Why do we need these? We should try to avoid lint exceptions as much as possible, and using no-multiline-disable is ... basically, not OK.

Reply:
The interface properties where in camel case so I had to add this line to avoid lint exceptions but after reading documentation I have defined camel case properties in a single quotes and deleted **no-multiline-disable** comment.

Reply:
OK, thanks.

FYI, though, I can't see your new code, and I realize there's one other comment from me remaining that still needs to be addressed. I suggest, in the future, writing draft responses to all comments first (using "Start a review"), then pushing your code, then finally using the "Review changes" button at the top right to submit all the comments at once so that they match the new code. (This is because reviewers generally get notified when responses are submitted, so at that point it would be good if the comments match the code.)

(I've updated the [corresponding wiki page](https://github.com/oppia/oppia/wiki/Make-a-pull-request#step-5-address-review-comments-until-all-reviewers-approve) to be clearer about this. -- /cc @U8NWXD for info.)

Reply:
I am sorry I should have pushed all changes before commenting on your suggestions. I will follow the above chronology in the future PR code reviews. Now coming to your [earlier suggestion ](https://github.com/oppia/oppia/pull/14063#discussion_r761715107) I have added the prefix part but I think there is some issues in the second proposed suggestion of thowing exception. 
I have uploaded a video that should give enough evidence that changing image data from its encoded form to a random string or number will throw an exception not in backend service but rather in **image-upload-helper**  , which is called in [ translation-model component ](https://github.com/oppia/oppia/blob/f74b606e8511cd4296b3c99aad37e53b66cca196/core/templates/pages/contributor-dashboard-page/modal-templates/translation-modal.component.ts#L443)  to convert all encoded images data to blob and return a list of Imagesdata. If the Imagesdata returns successfully without any exceptions then only it is passed to [translate-service](https://github.com/oppia/oppia/blob/f74b606e8511cd4296b3c99aad37e53b66cca196/core/templates/pages/contributor-dashboard-page/modal-templates/translation-modal.component.ts#L445) which then calls [backend-service.ts](https://github.com/oppia/oppia/blob/develop/core/templates/pages/contributor-dashboard-page/services/translate-text.service.ts#L224)

I think exception is raised on line 51.
https://github.com/oppia/oppia/blob/f74b606e8511cd4296b3c99aad37e53b66cca196/core/templates/services/image-upload-helper.service.ts#L51


https://user-images.githubusercontent.com/37933427/144746791-4f52b48f-b442-43d7-b9ea-f85a6d455ee7.mp4

So I think we should not handle exception in **backend-service** but rather catch in **translation-modal.component** and then probably put that on translation-modal.html.

Reply:
It might make more sense to comment in that thread, rather than this one.

I think we always need to handle errors in individual functions. When reading that function I had no knowledge of the other services, but functions should generally stand alone unless there's some clear guarantee that certain invariants hold. But in this case reader.result can be "anything", when looked at from the context of that function, so I think it needs to have proper error handling as well.